File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.ArrayList;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        
        if(cacheSize == 0) {
        	return cities.length * 5;
        }
              
        ArrayList<String> cacheList = new ArrayList<String>();
        
        for(int i = 0; i < cities.length; i++) {
        	String s = cities[i].toLowerCase();
        	
        	if(!cacheList.contains(s)) {
        		if(cacheList.size() == cacheSize) cacheList.remove(0); 
        		cacheList.add(s);
        		answer += 5;
        	}
        	else {
        		int index = cacheList.indexOf(s);
        		cacheList.remove(index);
        		cacheList.add(s);
        		answer += 1;
        	}
        }
        
        return answer;
    }
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    if cacheSize == 0:
        return len(cities) * 5

    answer = 0

    cache = []

    for city in cities:
        cityName = city.upper()
        is_incache = cityName in cache

        answer += 1 if is_incache else 5

        if len(cache) == cacheSize:
            if is_incache:
                del cache[cache.index(cityName)]
            else:
                del cache[0]    

        cache.append(cityName)

    return answer